arcgis-ios-sdk-samples/Route & Directions/Navigate route/NavigateRouteViewController.swift

```diff
@@ -0,0 +1,284 @@
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+import AVFoundation
+
+class NavigateRouteViewController: UIViewController {
+    /// The route result solved by the route task.
+    var routeResult: AGSRouteResult!
+    /// The original view point that can be reset to later on.
+    var defaultViewPoint: AGSViewpoint?
+    /// The graphic (with a dashed line symbol) to represent the route ahead.
+    var routeAheadGraphic: AGSGraphic!
+    /// The graphic to represent the route that's been traveled (initially empty).
+    var routeTraveledGraphic: AGSGraphic!
+    /// A list to keep track of directions solved by the route task.
+    var directionsList: [AGSDirectionManeuver] = []
+    /// The route tracker for navigation. Use delegate methods to update tracking status.
+    var routeTracker: AGSRouteTracker!
+    /// The route task to solve the route between stops, using the online routing service.
+    var routeTask: AGSRouteTask!
+    /// The initial location for the solved route.
+    var initialLocation: AGSLocation!
+    /// A formatter to format a time value into human readable string.
+    lazy var timeFormatter: DateComponentsFormatter = {
+        let formatter = DateComponentsFormatter()
+        formatter.allowedUnits = [.hour, .minute, .second]
+        formatter.unitsStyle = .full
+        return formatter
+    }()
+    /// An AVSpeechSynthesizer for text to speech.
+    lazy var speechSynthesizer = AVSpeechSynthesizer()
+    
+    /// The bar button item that initiates the create convex hull operation.
+    @IBOutlet weak var navigateButtonItem: UIBarButtonItem!
+    /// The bar button item that removes the convex hull as well as the MapPoints.
+    @IBOutlet weak var resetButtonItem: UIBarButtonItem!
+    /// The bar button item that recenters the map to navigation pan mode.
+    @IBOutlet weak var recenterButtonItem: UIBarButtonItem!
+    /// The 3-line label to display navigation status.
+    @IBOutlet weak var statusLabel: UILabel!
+    /// The map view managed by the view controller.
+    @IBOutlet weak var mapView: AGSMapView! {
+        didSet {
+            mapView.map = AGSMap(basemap: .navigationVector())
+            mapView.graphicsOverlays.add(makeRouteOverlay())
+        }
+    }
+    
+    /// A wrapper function for operations after the route is solved by an AGSRouteTask.
+    ///
+    /// - Parameters:
+    ///   - routeResult: The result from `AGSRouteTask.solveRoute(with:completion:)`.
+    ///   - error: The error from `AGSRouteTask.solveRoute(with:completion:)`.
+    func solveRouteCompletion(with routeResult: AGSRouteResult?, error: Error?) {
```

Comment:
This method name reads more like a command, like the method is going to be solving a route. Consider remaining it to something that removes that ambiguity, like `didSolveRoute`.